core/common/src/main/java/alluxio/util/UnderFileSystemUtils.java

```diff
@@ -67,5 +70,28 @@ public static void touch(final String path) throws IOException {
     os.close();
   }
 
+  /**
+   * Strips the folder suffix if it exists. This is a string manipulation utility and does not
+   * guarantee the existence of the folder. This method will leave keys without a suffix unaltered.
+   *
+   * @param key the key to strip the suffix from
+   * @return the key with the suffix removed, or the key unaltered if the suffix is not present
+   */
+  public static String stripFolderSuffixIfPresent(String key) {
```

Comment:
when I created this JIRA, there was no `CommonUtils.stripSuffixIfPresent` yet., and each under file system implement their own striping method in the identical way which is error prone. Since we now have `CommonUtils.stripSuffixIfPresent`,  it might be better just to use `CommonUtils.stripSuffixIfPresent` directly in each particular under file systems, without adding this util class which is anyway very thin. Does it work for you?